run_test.sh

```diff
@@ -26,6 +28,30 @@ case ${model} in
   *) echo "Unknown model: ${model}"; exit ;;
 esac
 
+./preprocessor.ESGF ${sformateddate} ${eformateddate} ${BCTABLE}
+test $? -eq 0 || exit 1
+
+# Running WPS and WRF
+HOMEDIR=`pwd`
+WRFDIR=${WRFDIR:-WRF}
+./util/deploy_WRF_CMake_binaries.sh ${WRFDIR}
+cd $HOMEDIR/$WRFDIR
+
+
+# Functions for updating namelists
+function format_date(){
```

Comment:
What is the purpose of this new function? The logic behind `set_nml` below has became quite more complex

Reply:
To separate sformateddate and eformateddate into year, month, day, and hour to correctly update start time and end time in namelist.wps and namlist.input. In the original version I had problems when testing WRF and WPS because dates in namelist.wps and namelist.input were taken directly from the template.

Reply:
Ok. But this function relies heavily on global variable definitions for input and output. It would be simpler to replace the call below (L75) by:
```bash
read year month day hour trash <<< `echo ${sformateddate} | tr '_T:-' '    '`
``` 
or similar (you'll still need to switch between s/eformatteddate)